src/main/java/at/hannibal2/skyhanni/features/chat/CompactJacobClaim.kt

```diff
@@ -0,0 +1,223 @@
+        " {4}§r§7§a(?<type>Jacob's|Carnival) Ticket §8x(?<amount>[\\d,]+)"
+    )
+
+    /**
+     * REGEX-TEST:     §r§7§81x §9Turbo-Cacti I Book
+     * REGEX-TEST:     §r§7§81x §9Turbo-Pumpkin I Book
+     * REGEX-TEST:     §r§7§84x §9Turbo-Wheat I Book
+     * REGEX-TEST:     §r§7§86x §9Turbo-Mushrooms I Book
+     * REGEX-TEST:     §r§7§81x §9Turbo-Warts I Book
+     */
+    private val bookPattern by patternGroup.pattern(
+        "book",
+        " {4}§r§7§8(?<amount>[\\d,]+)x §9Turbo-(?<crop>[^ ]+) I Book"
+    )
+
+    /**
+     * REGEX-TEST:     §r§7§8+§e8 §7gold medals
+     * REGEX-TEST:     §r§7§8+§e7 §7silver medals
+     * REGEX-TEST:     §r§7§8+§e5 §7bronze medals
+     */
+    private val medalsPattern by patternGroup.pattern(
+        "medals",
+        " {4}§r§7§8\\+§e(?<amount>[\\d,]+) §7(?<type>[^ ]+) medals?"
+    )
+
+    /**
+     * REGEX-TEST:     §r§8+§r§b2,293 Bits
+     */
+    private val bitsPattern by patternGroup.pattern(
+        "bits",
+        " {4}§r§8\\+§r§b(?<amount>[\\d,]+) Bits"
+    )
+    // </editor-fold>
+
+    private var inLoop = false
+    private var rewardSet = ContestRewardSet()
+    private val messageSet: MutableList<String> = mutableListOf()
+
+    private fun publishEvent() {
+        ContestRewardsClaimedEvent(rewardSet, messageSet).post()
+        rewardSet.reset()
+        messageSet.clear()
+    }
+
+    private val shorteningMap: Map<CropType, Pair<LorenzColor, String>> = mapOf(
+        CropType.CACTUS to Pair(LorenzColor.GREEN, "Ca"),
+        CropType.SUGAR_CANE to Pair(LorenzColor.GREEN, "Su"),
+        CropType.MELON to Pair(LorenzColor.GREEN, "Me"),
+
+        CropType.MUSHROOM to Pair(LorenzColor.RED, "Mu"),
+        CropType.NETHER_WART to Pair(LorenzColor.RED, "Ne"),
+
+        CropType.CARROT to Pair(LorenzColor.GOLD, "Ca"),
+        CropType.COCOA_BEANS to Pair(LorenzColor.GOLD, "Co"),
+        CropType.POTATO to Pair(LorenzColor.GOLD, "Po"),
+        CropType.PUMPKIN to Pair(LorenzColor.GOLD, "Pu"),
+        CropType.WHEAT to Pair(LorenzColor.GOLD, "Wh"),
+    )
+
+    private fun SkyHanniChatEvent.block(reason: String) {
+        messageSet.add(message)
+        blockedReason = reason
+    }
+
+    @HandleEvent(onlyOnIsland = IslandType.GARDEN)
+    fun onChat(event: SkyHanniChatEvent) {
+        if (!config.compactJacobClaim) return
+        val message = event.message
+        var eventDelay = 300.milliseconds
+
+        openingPattern.matchMatcher(message) {
+            inLoop = true
+            return event.block("compact_jacob_bulk_claim")
+        }
+
+        if (!inLoop) return
+
+        rewardsPattern.matchMatcher(message) {
+            return event.block("compact_jacob_bulk_claim")
+        }
+
+        val startingHash = rewardSet.hashCode()
+
+        ticketPattern.matchMatcher(message) {
+            val amount = group("amount").formatInt()
+            when (group("type")) {
+                "Jacob's" -> rewardSet = rewardSet.copy(jacobTickets = amount)
```

Comment:
why are we using copy all there and not reusing the same instance of rewards all the time?